backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/configuration/KeyspaceConfiguration.java

```diff
@@ -0,0 +1,98 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.cassandra.init.configuration;
+
+import java.util.Objects;
+
+import com.google.common.base.Preconditions;
+
+public class KeyspaceConfiguration {
+
+    public interface Builder {
+        @FunctionalInterface
+        interface RequireKeyspace {
+            RequireReplicationFactor keyspace(String name);
+        }
+
+        @FunctionalInterface
+        interface RequireReplicationFactor {
+            RequireDurableWrites replicationFactor(int replicationFactor);
+        }
+
+        @FunctionalInterface
+        interface RequireDurableWrites {
+            KeyspaceConfiguration durableWrites(boolean durableWrites);
+
+            default KeyspaceConfiguration disableDurableWrites() {
+                return durableWrites(false);
+            }
+        }
+    }
+
+    private static final String DEFAULT_KEYSPACE = "apache_james";
+    private static final int DEFAULT_REPLICATION_FACTOR = 1;
```

Comment:
Is it a sane default ?

Reply:
Yes. I don't ame at altering already choosen default values.

Also, we want things to work "by default" on a single node cassandra, which implies a replication factor of 1.

Reply:
Ok then, but IHMO it's the kind of thing where we don't want a default value.